linode/terraform-provider-linode
# Migrate `firewalldevice` package to Terraform Framework

## 📝 Description

This is to migrate `linode_firewall_device` from Terraform SDKv2 to Framework.

**This PR contains a proposed breaking change that converting time string into RFC3339 format (`timeObj.Format(time.RFC3339)`) from the Golang default format (`timeObj.String()`) in the previous SDKv2 resource.**

## ✔️ How to Test

### Automated Testing

`make testacc PKG_NAME="linode/firewalldevice"`

### TF Config for Manual Testing

```terraform
resource "linode_firewall_device" "my_device" {
  firewall_id = linode_firewall.my_firewall.id
  entity_id = linode_instance.my_instance.id
}

resource "linode_firewall" "my_firewall" {
  label = "my_firewall"

  inbound {
    label    = "http"
    action = "ACCEPT"
    protocol  = "TCP"
    ports     = "80"
    ipv4 = ["0.0.0.0/0"]
    ipv6 = ["::/0"]
  }

  inbound_policy = "DROP"
  outbound_policy = "ACCEPT"
}

resource "linode_instance" "my_instance" {
  label      = "my_instance"
  region     = "us-mia"
  type       = "g6-nanode-1"
}
```

- Build and apply this config with older Linode TF provider (`dev` branch) by `terraform apply`.
- Build and run `terraform plan` with the newer Linode TF provider (in this PR) to verify the plan is empty (no change is needed to be made) and everything works well.
- Destroy it by `terraform destroy` with the newer provider.
- Re-apply everything with the newer provider by `terraform apply`.
